spec/requests/topics/index_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'GET /topics', type: :request do
+  let(:user) { create :user }
+
+  subject do
+    get '/topics', headers: headers
+    data
+  end
+
+  before { create_list :topic, 3 }
+
+  it 'returns 3 topics' do
```

Comment:
You can test that the topics returned are the one that you have created by verifying its information. So you make sure you are not returning three times the same topic or whatever, in this case it's really simple, you are just doing Topic.all but take it into account for the future